From 7d0bd8b2cc4591c3b6e82a98536202eb525a6b01 Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Wed, 4 Mar 2015 14:20:53 +0800
Subject: [PATCH 279/284] ov680: move gpio2 to gpio1 and remove clk1

As BIOS could move original gpio2 to gpio1, we could still keep 2
gpios configure.

CLK1 is not used for ov680. remove it from driver.

Tracked-On: #H1503921753
Change-Id: I9ca5b632ce8deb81b132d7d64fd7611514d00cd9
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/44315
Reviewed-by: Jing Tao <jing.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Build: viefandr
Integrator: Wei Tao <wei.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/i2c/ov680.c               | 22 +++++-----------------
 1 file changed, 5 insertions(+), 17 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov680.c b/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
index 120b81d..2efc296 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
@@ -511,9 +511,9 @@ static int __gpio_ctrl(struct v4l2_subdev *sd, int on)
 #ifdef CONFIG_GMIN_INTEL_MID
 	if (on) {
 		/* fsa642 gpio */
-		ret = dev->platform_data->gpio2_ctrl(sd, 1);
+		ret = dev->platform_data->gpio1_ctrl(sd, 1);
 		if (ret)
-			goto gpio2_fail;
+			goto gpio1_fail;
 
 		/*
 		 * FIXME: setting gpio0 to 0 is not necessary for MOOR DR,
@@ -532,14 +532,14 @@ static int __gpio_ctrl(struct v4l2_subdev *sd, int on)
 		msleep(40);
 	} else {
 		dev->platform_data->gpio0_ctrl(sd, 0);
-		dev->platform_data->gpio2_ctrl(sd, 0);
+		dev->platform_data->gpio1_ctrl(sd, 0);
 	}
 
 	return 0;
 
 gpio0_fail:
-	dev->platform_data->gpio2_ctrl(sd, 0); /* fsa642 gpio */
-gpio2_fail:
+	dev->platform_data->gpio1_ctrl(sd, 0); /* fsa642 gpio */
+gpio1_fail:
 	dev_err(&client->dev, "failed to set gpio %s.\n", on ? "on" : "off");
 #endif
 	return ret;
@@ -560,18 +560,6 @@ static int __flisclk_ctrl(struct v4l2_subdev *sd, int on)
 		}
 	}
 
-#ifdef CONFIG_GMIN_INTEL_MID
-	if (dev->platform_data->flisclk1_ctrl) {
-		dev->platform_data->flisclk1_ctrl(sd, on);
-		if (ret) {
-			dev_err(&client->dev,
-				"%s - platform flisclk1 %s error.\n", __func__,
-				on ? "on" : "off");
-			if (on)
-				dev->platform_data->flisclk_ctrl(sd, 0);
-		}
-	}
-#endif
 	return ret;
 }
 
-- 
1.9.1

